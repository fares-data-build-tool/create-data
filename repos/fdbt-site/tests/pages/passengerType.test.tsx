import * as React from 'react';
import { shallow } from 'enzyme';
import PassengerType from '../../src/pages/passengerType';

describe('pages', () => {
    describe('operator', () => {
        it('should render correctly', () => {
            const tree = shallow(<PassengerType errors={[]} csrfToken="" savedGroups={[]} />);
            expect(tree).toMatchSnapshot();
        });
    });
});
