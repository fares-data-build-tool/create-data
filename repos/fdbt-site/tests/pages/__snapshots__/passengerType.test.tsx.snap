// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages operator should render correctly 1`] = `
<Component
  description="Passenger Type selection page of the Fares Data Build Tool"
  errors={Array []}
  title="Passenger Type - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/passengerType"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="passenger-type-page-heading"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="passenger-type-page-heading"
          >
            Select a passenger type
          </h1>
        </legend>
        <span
          className="govuk-hint"
          id="passenger-type-hint"
        >
          Relate the ticket(s) to a passenger type
        </span>
        <FormElementWrapper
          errorClass="govuk-radios--error"
          errorId="passenger-type-group"
          errors={Array []}
        >
          <div
            className="govuk-radios"
          >
            <div
              className="govuk-radios__item"
              key="group"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-group"
                name="passengerType"
                type="radio"
                value="group"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-group"
              >
                Group (more than one passenger)
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="adult"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-adult"
                name="passengerType"
                type="radio"
                value="adult"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-adult"
              >
                Adult
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="child"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-child"
                name="passengerType"
                type="radio"
                value="child"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-child"
              >
                Child
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="infant"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-infant"
                name="passengerType"
                type="radio"
                value="infant"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-infant"
              >
                Infant
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="senior"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-senior"
                name="passengerType"
                type="radio"
                value="senior"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-senior"
              >
                Senior
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="student"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-student"
                name="passengerType"
                type="radio"
                value="student"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-student"
              >
                Student
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="youngPerson"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-youngPerson"
                name="passengerType"
                type="radio"
                value="youngPerson"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-youngPerson"
              >
                Young Person
              </label>
            </div>
            <div
              className="govuk-radios__item"
              key="anyone"
            >
              <input
                className="govuk-radios__input"
                id="passenger-type-anyone"
                name="passengerType"
                type="radio"
                value="anyone"
              />
              <label
                className="govuk-label govuk-radios__label"
                htmlFor="passenger-type-anyone"
              >
                Anyone
              </label>
            </div>
          </div>
        </FormElementWrapper>
      </fieldset>
    </div>
    <InsetText
      text="More passenger types will become available soon"
    />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</Component>
`;
